### PR TITLE
feat: proxy service

### PR DESCRIPTION
**Pull request by Asmadek**
* implementation of proxy web worker
* model for working with proxy and proxy worker

### CODE REVIEW

File: src/renderer/features/proxies/workers/proxy-worker.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        }
+      }),
+    );
+  } catch (e) {
+    console.log(e);
+  }
+
+  return {
+    proxiesToAdd,
+    proxiesToRemove: proxies.filter((p) => !existedProxies.some((ep) => isEqual(p, ep))),
```
Comment (pgolovkin) [inline src/renderer/features/proxies/workers/proxy-worker.ts:110]: here should be: 
1. proxied accounts to add
2. proxied accounts to remove
3. proxy accounts to add
4. proxy accounts to remove

File: src/renderer/features/proxies/workers/proxy-worker.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+function initConnection(chain: Chain, connection: Connection) {
+  return new Promise((resolve) => {
+    if (!chain) {
+      return;
+    }
+
+    try {
+      let provider: ProviderInterface | undefined;
+
+      if (!connection || connection.connectionType === ConnectionType.AUTO_BALANCE) {
```
Comment (pgolovkin) [inline src/renderer/features/proxies/workers/proxy-worker.ts:34]: if network is disabled then proxy worker shouldn't run for it

**Comment by tuul-wq** [comment]
Task linked: [CU-86933z3gm Proxy service: Service for search for proxied accounts](https://app.clickup.com/t/86933z3gm)

File: src/renderer/features/proxies/workers/proxy-worker.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+          const needToAddProxyAccount = accounts[proxiedAccountId];
+          const doesProxyExist = proxies.some((oldProxy) => proxyWorkerUtils.isSameProxies(oldProxy, newProxy));
+
+          if (needToAddProxyAccount) {
+            if (!doesProxyExist) {
+              proxiesToAdd.push(newProxy);
+            }
+
+            deposits[proxiedAccountId] = {
```
Comment (tuul-wq) [inline src/renderer/features/proxies/workers/proxy-worker.ts:128]: Looks like we can add deposit only once
```javascript
if (needToAddProxyAccount || needToAddProxiedAccount) {
  deposits[proxiedAccountId] = {
     ...deposits[proxiedAccountId],
     [chainId]: proxyData[0][1].toHuman(),
  };
}
```

**Review by tuul-wq — COMMENTED** [review]
A couple of suggestions

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**